R/pkg/R/DataFrame.R

```diff
@@ -2624,6 +2624,15 @@ setMethod("except",
 setMethod("write.df",
           signature(df = "SparkDataFrame"),
           function(df, path = NULL, source = NULL, mode = "error", ...) {
+            if (!is.character(path) && !is.null(path)) {
+              stop("path should be charactor, null or omitted.")
+            }
+            if (!is.character(source) && !is.null(source)) {
+              stop("source should be charactor, null or omitted. It is 'parquet' by default.")
```

Comment:
strictly speaking, it's `spark.sql.sources.default` property, and when it is not set, then it is `parquet`